Doc/library/pyexpat.rst

```diff
@@ -196,6 +196,33 @@ XMLParser Objects
    :exc:`ExpatError` to be raised with the :attr:`code` attribute set to
    ``errors.codes[errors.XML_ERROR_CANT_CHANGE_FEATURE_ONCE_PARSING]``.
 
+.. warning::
+
+   Calling ``SetReparseDeferralEnabled(False)`` has security implications,
+   as detailed below; please make sure to understand these consequences
+   prior to using the ``SetReparseDeferralEnabled`` method.
+
+.. method:: xmlparser.SetReparseDeferralEnabled(enabled)
+
+   Expat 2.6.0 introduced a security mechanism called "reparse deferral"
+   where instead of causing denial of service through quadratic runtime
+   from reparsing large tokens, reparsing of unfinished tokens is now delayed
+   by default until a sufficient amount of input is reached.
+   Due to this delay, registered handlers may — depending of the sizing of
+   input chunks pushed to Expat — no longer be called right after pushing new
+   input to the parser.  Where immediate feedback and taking over responsiblity
+   of protecting against denial of service from large tokens are both wanted,
+   calling ``SetReparseDeferralEnabled(False)`` disables reparse deferral
+   for the current Expat parser instance, temporarily or altogether.
+   Calling ``SetReparseDeferralEnabled(True)`` allows re-enabling reparse
+   deferral.
+
```

Comment:
Add `versionadded` directives for all new methods. Since they are purposed to be backported, I think that you can use `.. versionadded:: 3.12.3`. In backports to 3.11 and earlier it will be changed.

Reply:
@serhiy-storchaka I already had that on the radar for later today, will do. I'm unsure how 3.12 should be special here, I'd assume we start out with 3.13 and then adjust during backports for all other 3.x branches. Could you help me understand how 3.12.3 would help over 3.13 when targetting `main` here?

Reply:
@serhiy-storchaka PS: 4x `.. versionadded:: 3.13` added now for a start

Reply:
This feature will be supported in the continuous interval "3.12.3-" of Python releases. If the user sees `versionadded:: 3.12.3`, they can understand that in can be used in 3.12 (unless they have not updated it) without checking the 3.12 documentation. It is also a hint that it could be backported in 3,11 and older versions. If the user sees `versionadded:: 3.13`, there is no reason to suppose that this feature exist in older releases.

It may be only my personal opinion, but I think that this variant is better. But 3.13 is a good start, and we can discuss this later.